ui/library/test/context/ContextProvider.test.tsx

```diff
@@ -9,6 +9,21 @@ import { IScrollContext, ScrollContext } from '../../src/context/ScrollContext';
 import { ITransformContext, TransformContext } from '../../src/context/TransformContext';
 import { IUiContext, UiContext } from '../../src/context/UiContext';
 
+class MockIntersectionObserver {
+  constructor(onChange) {
+    this.__onChange = onChange;
+  }
+  observe() {
+    return null;
+  }
+  disconnect() {
+    return null;
+  }
+  unobserve() {
+    return null;
+  }
+}
+
```

Comment:
Since you are using `MockIntersectionObserver` in all 3 test files, maybe it is a good idea to create a file for this class.

Reply:
@yensung not a bad idea i can do that